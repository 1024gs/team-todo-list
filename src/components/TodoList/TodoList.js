import React from 'react';
import './TodoList.scss';
import TodoItem from '../TodoItem/TodoItem';

const TodoList = ({items, getItemId, onAdd, onToggle, onDelete}) => {
  return (
    <div className="TodoList">
      {items.map((item) => (
        <TodoItem
          item={item}
          key={getItemId(item)}
          getItemId={getItemId}
          onToggle={onToggle}
          onDelete={onDelete}
        />
      ))}
    </div>
  );
};

export default TodoList;
