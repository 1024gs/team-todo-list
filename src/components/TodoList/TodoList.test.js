import React from 'react';
import TodoList from './TodoList';
import sinon from 'sinon';
import {shallow} from 'enzyme';

const setup = (propOverrides) => {
  const props = Object.assign(
    {
      items: [
        {id: 1, title: 'Todo 1'},
        {id: 2, title: 'Todo 2', isCompleted: true}
      ],
      getItemId: (x) => x.id,
      onAdd: sinon.spy(),
      onEdit: sinon.spy(),
      onToggle: sinon.spy(),
      onDelete: sinon.spy()
    },
    propOverrides
  );
  const wrapper = shallow(<TodoList {...props} />);

  return {
    props,
    wrapper
  };
};

describe('TodoList', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
