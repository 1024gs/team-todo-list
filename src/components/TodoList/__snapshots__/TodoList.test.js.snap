// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders 1`] = `
<div
  className="TodoList"
>
  <TodoItem
    getItemId={[Function]}
    item={
      Object {
        "id": 1,
        "title": "Todo 1",
      }
    }
    key="1"
    onDelete={[Function]}
    onToggle={[Function]}
  />
  <TodoItem
    getItemId={[Function]}
    item={
      Object {
        "id": 2,
        "isCompleted": true,
        "title": "Todo 2",
      }
    }
    key="2"
    onDelete={[Function]}
    onToggle={[Function]}
  />
</div>
`;
